Mathlib/Algebra/Ring/Divisibility/Lemmas.lean

```diff
@@ -0,0 +1,89 @@
+/-
+Copyright (c) 2023 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import Mathlib.Algebra.Ring.Divisibility.Basic
+import Mathlib.Data.Nat.Choose.Sum
+
+/-!
+# Lemmas about divisibility in rings
+
+# Main results:
+ * `dvd_smul_of_dvd`: stating that `x ∣ y → x ∣ m • y` for any scalar `m`.
+ * `Commute.pow_dvd_add_pow_of_pow_eq_zero_right`: stating that if `y` is nilpotent then
+   `x ^ m ∣ (x + y) ^ p` for sufficiently large `p` (together with many variations for convenience).
```

Comment:
This heading shouldn't be the same level as the main heading:

```suggestion
## Main results:

* `dvd_smul_of_dvd`: stating that `x ∣ y → x ∣ m • y` for any scalar `m`.
* `Commute.pow_dvd_add_pow_of_pow_eq_zero_right`: stating that if `y` is nilpotent then
  `x ^ m ∣ (x + y) ^ p` for sufficiently large `p` (together with many variations for convenience).
```

(also some minor spacing things that markdown doesn't really care about)